### PR TITLE
Add redis

### PR DESCRIPTION
**Pull request by kalefranz**

### CODE REVIEW

File: recipes/redis/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+test:
+  requires:
+    - tk 8.5*
+  commands:
+    - "cd {{ SRC_DIR }} && tclsh8.5 tests/test_helper.tcl"
+
+about:
+  home: http://redis.io
+  license: BSD
```
Comment (msarahan) [inline recipes/redis/meta.yaml]: 2 clause or 3?

File: recipes/redis/build.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+unset ARCH  # https://github.com/redis/hiredis-rb/issues/2
+make distclean
+make PREFIX=$PREFIX install
```
Comment (msarahan) [inline recipes/redis/build.sh]: Is there a make check/test?

Comment (kalefranz) [inline recipes/redis/build.sh]: No, that's the `tclsh8.5 tests/test_helper.tcl`

File: recipes/redis/meta.yaml (hunk 1 above)
Comment (kalefranz) [inline recipes/redis/meta.yaml]: 3.  Should this field's value be `BSD 3-Clause`, `3-Clause BSD`, or something else?

Comment (msarahan) [inline recipes/redis/meta.yaml]: I see both.  I don't think it matters much.

**Comment by kalefranz** [comment]
Rebased against master to hopefully fix the windows error.  (Windows builds should be being skipped anyway.)

**Comment by kalefranz** [comment]
@msarahan @pelson Any hints on why the windows build is failing here? I'm including

    build:
      skip: true  # [win]

in the `meta.yaml` file...

**Comment by Korijn** [comment]
This package has the same name as the python-client package for redis; that's going to be confusing.

**Comment by ocefpaf** [comment]
I would like to rename the Python client to python-redis and leave this one as is.

**Comment by Korijn** [comment]
I think others have adopted the name `redis-py`... might want to mirror that instead?

**Comment by kalefranz** [comment]
@ocefpaf Any hints on what we need to do here to be all green?  I've tried to disable the windows builds, but apparently I haven't done it quite right.

**Comment by msarahan** [comment]
I think you should just ignore it.  It's failing to parse the recipe before it even gets to the selector phase.  Make sure to add [skip appveyor] to your commit messages on this repo, but otherwise it's OK.  Do we need to rename any other feedstocks before merging this PR?

**Comment by kalefranz** [comment]
`[skip appveyor]`... Cool!

**Comment by jakirkham** [comment]
Not sure how I feel about this mucking with the system root. Is this really necessary or can one just specify a prefix to sort these files to and use `$PREFIX` paths by default?

**Comment by ocefpaf** [comment]
> @ocefpaf Any hints on what we need to do here to be all green? 

Apologies, I am travelling and my response time is slow. This issues seems to be resolved. We can discuss the python bindings later as this one doesn't need to be renamed (right?).

I am :+1: for the merge. Let's just what we can do about https://github.com/jakirkham before merging it.

File: recipes/redis/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+build:
+  number: 0
+  skip: true  # [win]
+  script_env:
+    - "MAJOR_MINOR_VERSION={{ '.'.join(version.split('.')[0:2]) }}"
+
+test:
+  requires:
+    - tk 8.5*
```
Comment (jakirkham) [inline recipes/redis/meta.yaml]: `8.5*` -> `8.5.*`

**Comment by ocefpaf** [comment]
@kalefranz is this good to go? There is a minor comment on the `tk` pinning and I believe it is good to merge.

File: recipes/redis/meta.yaml (hunk 1 above)
Comment (pelson) [inline recipes/redis/meta.yaml]: Agreed. We have generally leant towards BSD 3-Clause, but it isn't enforced as a standard (currently).

**Comment by pelson** [comment]
When it comes to it, I would prefer the python client to be named `python-redis`, as per Fedora and Debian - unfortunately we wont be in a position to do that, since there already exists a `redis-py` in defaults.
As has been stated elsewhere, we are currently bound to mirroring the name naming convention between defaults and conda-forge.

In sort, `redis-py` it is.


> Not sure how I feel about this mucking with the system root.

I agree completely with this. Even if we had appropriate permissions (i.e. root) I don't think we can modify a user's ``/var`` and ``/etc/`` in good faith. Is that a show-stopper for this recipe @kalefranz? Is the problem that this redis package is really the package + configuration for a sensible default setup?

**Comment by Korijn** [comment]
@andymccurdy just giving you a headsup of the work here :) also, I wanted to check with you, just to be sure: `redis-py` is the official package name of the python client, correct?

**Comment by andymccurdy** [comment]
redis-py is the official name of the Python client. I named it after the official redis-rb, the official Ruby client.

File: recipes/redis/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  commands:
+    - "cd {{ SRC_DIR }} && tclsh8.5 tests/test_helper.tcl"
```
Comment (jakirkham) [inline recipes/redis/meta.yaml]: Maybe this should be in `build.sh`. Seems more natural than switching back to the source directory here.

File: recipes/redis/supervisord-startup-redis [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    mkdir -pm 755 $PREFIX/var/run/redis
+    chown redis:redis $PREFIX/var/run/redis
+    mkdir -p /var/run
+    ln -s $PREFIX/var/run/redis /var/run/redis
+
+    exec runuser redis -c '$PREFIX/bin/redis-server $PREFIX/etc/redis/redis.conf'
+else
+    exec $PREFIX/bin/redis-server $PREFIX/etc/redis/redis.conf
+fi
+
```
Comment (jakirkham) [inline recipes/redis/supervisord-startup-redis]: nit: please remove extra newline.

**Comment by sodre** [comment]
@kalefranz 
I would like to continue this work to bring redis + supervisord into condo-forge. Do you have any objections?

@jakirkham & @pelson  if @kalefranz is okay, can you ensure a quick review/approval process?

**Comment by jakirkham** [comment]
@sodre, assuming you are still interested in this, I would just submit a new PR. We can start off using this content if it makes it easier to get going. Can also integrate any comments from people interested in this recipe in the process.

**Pull request closed without merging**